File: src/main/scala/NextPrime.scala
import scala.io.StdIn.readLine
import util.control.Breaks.{breakable, break}
object NextPrime extends App {
  var num = readLine().trim.toInt
  if(num <= 2){
    println(s"Prime number = 2")
  }else{
    checkPrime(num, true)
  }

  def checkPrime(pnum: Int, pprime: Boolean): Unit ={
    var prime = pprime
    if(pnum%2 == 0)checkPrime(pnum+1,true)
    else{
      breakable{
        for(i <- 3 to pnum/2 by 2){
          if(pnum%i == 0){
            prime = false
            break()
          }
        }
      }
      if(prime && num != pnum && pnum%2 != 0){
        println(s"Next prime number after $num = $pnum")
      }else if(prime && pnum%2 != 0){
        println(s"$num is a prime number")
      }else{
        checkPrime(pnum+1,true)
      }
    }
  }//end checkPrime
}

File: PythonSolutions/NextPrime.py


def checkifprime(pnum, isprime=True):

    prime = isprime
    if pnum % 2 == 0:
        checkifprime(pnum+1)
    else:
        for i in range(3, pnum//2, 2):
            if pnum % i == 0:
                prime = False
                break
        if prime and number != pnum:
            print('next prime after {} = {}'.format(number, pnum))
        elif prime:
            print('{} is a prime number.'.format(number))
        else:
            checkifprime(pnum+1)


number = int(input('Insert the number to check> '))

if number <= 2:
    print('Next prime = 2')
else:
    checkifprime(number, True)
